### PR TITLE
Bump skydns godeps to latest

### PR DESCRIPTION
**Pull request by girishkalele**
Update Godeps for github.com/skynetservices/skydns and miekg/dns.

Bump kubedns version to 1.6 with latest skynetservices/skydns code

Built kube-dns for all architectures and pushed containers to gcr.io.

### CODE REVIEW

**Comment by girishkalele** [comment]
Big ugly Godeps change - first commit only interesting file is Godeps.json.

Second commit bumps the kubedns tag.

**Comment by girishkalele** [comment]
@roberthbailey 

Background is that we didn't want to pickup latest skydns code before 1.3 - this commit should give us plenty of soak time before 1.4.

**Comment by roberthbailey** [comment]
This looks like a pretty large change, and I won't have time to review it for a couple of days.

File: cluster/saltbase/salt/kube-dns/skydns-rc.yaml.base [hunk 1]
```diff
@@ -21,27 +21,27 @@
...
       labels:
         k8s-app: kube-dns
-        version: v17
+        version: v18
         kubernetes.io/cluster-service: "true"
     spec:
       containers:
       - name: kubedns
-        image: gcr.io/google_containers/kubedns-amd64:1.5
+        image: gcr.io/google_containers/kubedns-amd64:1.6
```
Comment (thockin) [inline cluster/saltbase/salt/kube-dns/skydns-rc.yaml.base]: have you built and pushed this yet?

**Comment by thockin** [comment]
LGTM

File: cluster/saltbase/salt/kube-dns/skydns-rc.yaml.base (hunk 1 above)
Comment (girishkalele) [inline cluster/saltbase/salt/kube-dns/skydns-rc.yaml.base]: Yes, pushed for all architectures. Tested amd64.

**Comment by girishkalele** [comment]
@k8s-bot test this #IGNORE

Could not find a matching flake for this failure.

**Pull request merged**